### PR TITLE
Handle errors during app bootstrap

### PR DESCRIPTION
**Pull request by garethbowen**
API was throwing uncaught promise rejections which caused the
client to hang indefinitely waiting for a response during
bootstrapping.

In addition the client wasn't correctly detecting errors during
`fetch`.

medic/cht-core#6084

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/tests/mocha/unit/purger.spec.js [hunk 1]
```diff
@@ -64,21 +64,31 @@ describe('Purger', () => {
       });
     });
 
-    it('should throw fetch errors', () => {
```
Comment (dianabarsan) [inline webapp/tests/mocha/unit/purger.spec.js]: Could we keep this test and just add the new one where the server returns 500? Fetch throws when not connected, which is much more likely to occur frequently than API failing.

File: webapp/tests/mocha/unit/purger.spec.js [hunk 2]
```diff
@@ -64,21 +64,31 @@ describe('Purger', () => {
...
+        })
+        .catch(err => {
+          chai.expect(err.message).to.equal('Error fetching purge data: {"code":500,"error":"Server error"}');
+          done();
+        });
     });
   });
 
   describe('checkpoint', () => {
-    it('should throw fetch errors', () => {
```
Comment (dianabarsan) [inline webapp/tests/mocha/unit/purger.spec.js]: Could we keep this test and just add the new one where the server returns 500? Fetch throws when not connected, which is much more likely to occur frequently than API failing.

Comment (garethbowen) [reply webapp/tests/mocha/unit/purger.spec.js]: Absolutely. Done.

File: webapp/tests/mocha/unit/purger.spec.js (hunk 1 above)
Comment (garethbowen) [reply webapp/tests/mocha/unit/purger.spec.js]: Absolutely. Done.

**Pull request merged**